### PR TITLE
[WIP] Submodules rework

### PR DESCRIPTION
**Pull request by batiati**
### This PR:

- Moves the C client to `/clients`.
- Removes the `tigerbeetle` submodule/symlink from the Java and Go client.
- Improves `build.zig` to build and update the clients instead of using sh/bat build scripts.

### Usage:

**Java**: cross-compiles all JNI shared libraries, placing them in the `JAR` folder to be packaged:
```bash
./zig/zig build java_client -Drelease-safe
```

**Go**: cross-compiles all static libraries placing them in the `pkg/native` folder, and also updates the `tb_client.h` header:
```bash
 ./zig/zig build go_client -Drelease-safe
```

### Pending work - WIP

- DotNet client.
- Node client.
- CI
- README and build instructions.

### Open to suggestions:

Should we invoke language-specific commands from `build.zig` ?
For example, the command to build the Java client should be:

**Option 1:** Zig builds only the native code, and language-specific tasks must be invoked separately:

```bash
./zig/zig build java_client -Drelease-safe
cd src/clients/java/src/tigerbeetle-java
mvn package
```
**Option 2:**  Zig builds the whole package. Just calling `./zig/zig build java_client -Drelease-safe` also invokes Java/Maven specific commands.

```bash
./zig/zig build java_client -Drelease-safe
```

The same logic can be applied to running tests, executing samples, benchmarks, etc.

**Option 3:** Rely on language-specif toolchains for building the clients. Just calling `mvn package` also invokes `zig build`.

```bash
cd src/clients/java/src/tigerbeetle-java
mvn package
```

### CODE REVIEW

File: build.zig [hunk 1]
```diff
@@ -404,3 +406,110 @@ fn link_tracer_backend(
...
+        const lib = b.addStaticLibrary("tb_client", "src/clients/c/tb_client.zig");
+        lib.setMainPkgPath("src");
+        lib.setTarget(cross_target);
+        lib.setBuildMode(mode);
+        lib.pie = true;
+        lib.bundle_compiler_rt = true;
+
+        if (cross_target.os_tag.? != .windows) {
+            lib.linkLibC();
+        }
```
Comment (kprotty) [inline build.zig:346]: We can always linkLibC() on windows now (`builtin.link_libc` is used to [know when to export](https://github.com/tigerbeetledb/tigerbeetle/blob/main/src/c/tb_client.zig#L57-L65) the tb_client functions)

**Review by kprotty — COMMENTED** [review]
Nice work! Really cleans up the build process for the clients. One last tidbit for C header generation on windows.

File: build.zig (hunk 1 above)
Comment (batiati) [reply build.zig:346]: Thanks for the tip! That's why the Java CI is failing right now on Windows!

**Review by kprotty — DISMISSED** [review]

**Comment by kprotty** [comment]
Need to rebase on main + address the CI issues though

**Comment by batiati** [comment]
superseded by #320

**Pull request closed without merging**